Tools/TransformTool/TransformTool.cs

```diff
@@ -436,6 +436,9 @@ public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeCon
 
                     if (leftInput.select.wasJustReleased)
                     {
+                        if (rightInput != null && rightInput.select.wasJustReleased)
+                            m_Scaling = false;

```

Comment:
This was the fix for the "Main Menu Hidden" bug. Apparently it happens pretty frequently that both trigger releases happen in the same frame.